packages/ping_federate/data_stream/admin/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,255 @@
+---
+description: Pipeline for processing Admin logs.
+processors:
+  - set:
+      field: ecs.version
+      tag: set_ecs_version
+      value: 8.16.0
+  - set:
+      field: observer.vendor
+      tag: set_observer_vendor
+      value: "Ping Identity"
+  - set:
+      field: observer.product
+      tag: set_observer_product
+      value: PingFederate
+  - rename:
+      field: _conf.tz_offset
+      target_field: event.timezone
+      if: ctx._conf?.tz_offset != null
+      tag: rename_tz_offset
+      ignore_missing: true
+  - rename:
+      field: message
+      tag: rename_message_to_event_original
+      target_field: event.original
+      ignore_missing: true
+      description: Renames the original `message` field to `event.original` to store a copy of the original message. The `event.original` field is not touched if the document already has one; it may happen when Logstash sends the document.
+      if: ctx.event?.original == null
+  - remove:
+      field: message
+      tag: remove_message
+      ignore_missing: true
+      description: The `message` field is no longer required if the document has an `event.original` field.
+      if: ctx.event?.original != null
+  - grok:
+      field: event.original
+      patterns:
+        - '%{DATA:ping_federate.admin.timestamp}\s\|\s%{WORD:ping_federate.admin.user}\s\|\s(%{DATA:ping_federate.admin.roles})?\s\|\s(%{IP:ping_federate.admin.ip})?\s\|\s(%{DATA:ping_federate.admin.event.detail_id})?\s\|\s(%{WORD:ping_federate.admin.component})\s\|\s(%{WORD:ping_federate.admin.event.type})?\s\|\s(%{GREEDYDATA:ping_federate.admin.message})?'
+      if: ctx.event?.original != null
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - set:
+      field: event.kind
+      value: event
+      tag: set_event_kind
+  - append:
+      field: event.category
+      value: [authentication]
+      if: ctx.ping_federate?.admin?.event?.type != null
+      tag: append_event_category_authentication
+  - append:
+      field: event.category
+      value: [session]
+      if: ctx.ping_federate?.admin?.event?.type != null && ctx.ping_federate?.admin?.event?.type.contains('SESSION')
+      tag: append_event_category_session
+  - append:
+      field: event.type
+      value: [info]
+      tag: append_event_type_info
+      if: >-
+        ctx.ping_federate?.admin?.event?.type != null && (
+          ctx.ping_federate?.admin?.event?.type == 'MODIFY' ||
```

```suggestion
          ctx.ping_federate.admin.event.type == 'MODIFY' ||
```
Once checked for not-null, `?` can be ignored.

Same for other processors in both datastreams.